CRM/Report/Form/Contact/Detail.php

```diff
@@ -975,4 +976,44 @@ public function alterComponentDisplay(&$componentRows) {
     }
   }
 
+  /**
+   * Function to alter from clause by adding checks for permissoned financial type.
+   *
+   * @param string $component
+   *
+   * @return string
+   */
+  public function addPermissionedFromClause($component) {
+    if (is_null($this->financialTypes)) {
+      return NULL;
+    }
+    $financialTypes = 0;
+    if (!empty($this->financialTypes)) {
+      $financialTypes = implode(',', array_keys($this->financialTypes));
+    }
+    $clause = "financial_type_id IN ({$financialTypes})";
+    $whereClause = NULL;
+
+    switch ($component) {
+      case 'participant_civireport':
+      case 'membership_civireport':
+        $tableName = str_replace('_civireport', '', $component);
+        $alias = $this->_aliases["civicrm_{$tableName}"];
+        if ($component == 'membership_civireport') {
```

Comment:
Do we need this join? Generally we try to resolve membership type by pseudoconstant